clang/test/Driver/sycl.c

```diff
@@ -43,6 +43,11 @@
 // COMBINED: "-triple" "spir64-unknown-{{.*}}-sycldevice"{{.*}} "-fsycl-is-device"{{.*}} "-emit-llvm-bc"
 // TEXTUAL: "-triple" "spir64-unknown-{{.*}}-sycldevice{{.*}}" "-fsycl-is-device"{{.*}} "-emit-llvm"
 
+/// Verify that the sycl header directory is before /usr/include
+// RUN: %clangxx -### -fsycl-device-only %s 2>&1 | FileCheck %s --check-prefix=HEADER_ORDER
+// RUN: %clangxx -### -fsycl %s 2>&1 | FileCheck %s --check-prefix=HEADER_ORDER
+// HEADER_ORDER-NOT: clang{{.*}} "/usr/include"{{.*}} "-internal-isystem" "{{.*}}bin{{[/\\]+}}..{{[/\\]+}}include{{[/\\]+}}
```

Comment:
Do we have a test for SYCL headers' presence?
Also, is the `*/bin/../include` structure guaranteed? E.g. can imagine the case where `make install` was run instead of `*PATH` envvars' setting.

Reply:
Not sure if we have a test for the actual presence of the headers, but there are a number of tests that rely on them, and many tests have explicit inclusion (see https://github.com/intel/llvm/commit/39501f6bca9651777959ed8d96b2a326d03b2742)

As for the `bin/../include` structure, that is based on common clang install structure and the location of the driver.  It is what is used for all other checks of the header so I'm just continuing that expectation.